### ISSUE TITLE
Implementar rota que prediz os valores de Water_Need no context de culture

### ISSUE THREAD

**Issue by felipealencar** [lead-ifal/pc2i-platform#132]
Após a importação dos dados da cultura de milho (Wheat), ao chamar a nova rota `cultures_bp.route("/<culture_id>/water-need", methods=["GET"])(CultureController.predict)`, deverá ser criado o método `predict` em `CultureController`.

Esse método deverá
1. Carregar todos os dados importados para os sensores associados à cultura de milho;
2. Carregar todos os dados importados para o contexto da cultura de milho;
3. Agrupar em um só dataframe todas as colunas de ambos os dados (observe que as células das colunas de contexto ficarão com valores repetidos, e lembre de verificar se o dia de contexto bate com o timestamp da leitura dos sensores);
4. Com esse dataframe agrupado, normalize seus valores (exceto para as colunas de timestamp/date:
```
# normalize o dataset
scaler = MinMaxScaler(feature_range=(0, 1))
dataset = scaler.fit_transform(dataset)
```
5. Tenha certeza que seus dados estão ordenados em ordem crescente de data;
6. Separe o dataset em um conjunto de dados e em um conjunto de treinamento:
````
train_size = int(len(dataset) * 0.67)
test_size = len(dataset) - train_size
train, test = dataset[0:train_size,:], dataset[train_size:len(dataset),:]
print(len(train), len(test))
```
7. Crie um novo conjunto de dados com o que chamamos de _lookback_:
```
# converte um vetor de valores em uma matriz de dataset
def create_dataset(dataset, look_back=1):
	dataX, dataY = [], []
	for i in range(len(dataset)-look_back-1):
		a = dataset[i:(i+look_back), 0]
		dataX.append(a)
		dataY.append(dataset[i + look_back, 0])
	return np.array(dataX), np.array(dataY)

# colocando X como um instante presente, e Y  como um instante futuro
look_back = 1
trainX, trainY = create_dataset(train, look_back)
testX, testY = create_dataset(test, look_back)

# readequando o dataset pra que ele tenha um formato [samples, time steps, features]
trainX = np.reshape(trainX, (trainX.shape[0], 1, trainX.shape[1]))
testX = np.reshape(testX, (testX.shape[0], 1, testX.shape[1]))
```
9. Na mesma função do controlador, crie o modelo LSTM:
```
model = Sequential()
model.add(LSTM(4, input_shape=(1, look_back)))
model.add(Dense(1))
model.compile(loss='mean_squared_error', optimizer='adam')
model.fit(trainX, trainY, epochs=100, batch_size=1, verbose=2)
```
10. Faça as predições:
```
trainPredict = model.predict(trainX)
testPredict = model.predict(testX)

trainPredict = scaler.inverse_transform(trainPredict)
trainY = scaler.inverse_transform([trainY])
testPredict = scaler.inverse_transform(testPredict)
testY = scaler.inverse_transform([testY])
# Calcula a acurácia
trainScore = np.sqrt(mean_squared_error(trainY[0], trainPredict[:,0]))
print('Train Score: %.2f RMSE' % (trainScore))
testScore = np.sqrt(mean_squared_error(testY[0], testPredict[:,0]))
print('Test Score: %.2f RMSE' % (testScore))
```
11. Exiba os resultados (talvez seja preciso adaptar o plot para ser exibido como HTML):
```

trainPredictPlot = np.empty_like(dataset)
trainPredictPlot[:, :] = np.nan
trainPredictPlot[look_back:len(trainPredict)+look_back, :] = trainPredict

testPredictPlot = np.empty_like(dataset)
testPredictPlot[:, :] = np.nan
testPredictPlot[len(trainPredict)+(look_back*2)+1:len(dataset)-1, :] = testPredict

plt.plot(scaler.inverse_transform(dataset))
plt.plot(trainPredictPlot)
plt.plot(testPredictPlot)
plt.show()
```

Você pode usar esse material como referência, mas ele faz a predição de passageiros por dia. A diferença é que estamos predizendo a quantidade de água a ser irrigada: https://machinelearningmastery.com/time-series-prediction-lstm-recurrent-neural-networks-python-keras/.
